Content.Server/Database/ServerDbBase.cs

```diff
@@ -1016,6 +1016,79 @@ public async Task<int> CountAdminLogs(int round)
             await using var db = await GetDb();
             return await db.DbContext.AdminLog.CountAsync(log => log.RoundId == round);
         }
+        #endregion
+
+        #region Support Logging
+
+        /// <summary>
+        /// Retrieves a SupportExchange based on the round number, target user (Guid), and server id.
+        /// </summary>
+        public async Task<SupportExchange?> GetSupportExchangeAsync(int supportRound, Guid supportTarget)
+        {
+            await using var db = await GetDb();
+            return await db.DbContext.SupportExchanges
+                .Include(e => e.SupportMessages)
+                .SingleOrDefaultAsync(e => e.SupportRound == supportRound &&
+                                          e.SupportTargetPlayer == supportTarget);
+        }
+
+        /// <summary>
+        /// Adds a new SupportExchange to the database.
+        /// </summary>
+        public async Task<int> AddSupportExchangeAsync(SupportExchange exchange)
+        {
+            await using var db = await GetDb();
+            db.DbContext.SupportExchanges.Add(exchange);
+            await db.DbContext.SaveChangesAsync();
+            return exchange.SupportExchangeId;
+        }
+
+        /// <summary>
+        /// Adds a new SupportMessage to the database within an existing SupportExchange.
+        /// </summary>
+        public async Task AddSupportMessageAsync(SupportMessage supportMessage)
+        {
+            await using var db = await GetDb();
+
+            await using var transaction = await db.DbContext.Database.BeginTransactionAsync();
+
+            try
+            {
+                var maxMessageId = await db.DbContext.SupportMessages
+                    .Where(m => m.SupportExchangeId == supportMessage.SupportExchangeId)
+                    .OrderByDescending(m => m.SupportMessageId)
+                    .Select(m => m.SupportMessageId)
+                    .FirstOrDefaultAsync();
+
+                supportMessage.SupportMessageId = maxMessageId + 1;
+
+                db.DbContext.SupportMessages.Add(supportMessage);
+
+                await db.DbContext.SaveChangesAsync();
+
+                await transaction.CommitAsync();
+            }
+            catch (Exception e)
+            {
+                await transaction.RollbackAsync();
```

Comment:
You don't need to manually roll back the transaction, that's default behavior when a transaction object is disposed.